plugins/repository-gcs/src/main/java/org/elasticsearch/repositories/gcs/Crc32C.java

```diff
@@ -0,0 +1,109 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+package org.elasticsearch.repositories.gcs;
+
+import org.apache.lucene.util.BytesRef;
+import org.apache.lucene.util.BytesRefIterator;
+import org.elasticsearch.common.Numbers;
+import org.elasticsearch.common.bytes.BytesReference;
+
+import java.io.IOException;
+import java.util.Base64;
+
+/**
+ * CRC32C implementation to compute a checksum for resumable GCS uploads.
+ * TODO: Move to java.util.zip.CRC32C instead of this custom implementation once we no longer need to support JDK-8
+ */
+public final class Crc32C {
```

Comment:
Google recommends using crc32 over md5 so I went with this (MD5 isn't available for all objects, and while it seems that we never create an object that doesn't have an md5 this seemed safer).

Reply:
Converting to draft for a bit until I had the chance to double check the above assumption. Maybe MD-5 and thus keeping it simple works for us in all cases after all ...